Low-level IP, TCP, TLS/SSL errors on GitLab connection cause fatal crash

**Describe the bug**

GitLab code in CreateRelease() doesn't handle low-level IP, TCP, TLS/SSL errors properly and can attempt to dereference nil in resp.StatusCode. I'll do a PR with a trivial fix.
 
**To Reproduce**

Steps to reproduce the behavior:

```yaml
builds:
- goos:
    - linux
  goarch:
    - amd64
archive:
gitlab_urls:
  api: https://127.0.0.1/api/v4/
  download: https://127.0.0.1
  skip_tls_verify: false
```

**Expected behavior**

Printing out proper net/tls/x509/etc. error and exiting properly.

**Environment (please complete the following information):**

- OS: [Linux]
- OS version: [4.19.0-0.bpo.5-amd64]
- GoReleaser Version [0.113.0]

**Additional context**

Crashes on rereferencing nil in resp.StatusCode in internal/client/gitlab.go:71 whenever there is a low-level connectivity issue as resp doesn't have to be non-nil.

```
      • GITHUB/GITLAB RELEASES
         • creating or updating release repo= tag=v0.0.0
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe947cc]

goroutine 8 [running]:
github.com/goreleaser/goreleaser/internal/client.(*gitlabClient).CreateRelease(0xc0000103c8, 0xc00035a000, 0xc0005e1000, 0xff2, 0xc0005e1000, 0xff2, 0x0, 0x0)
        /root/work/log_rate/goreleaser/internal/client/gitlab.go:71 +0x38c
github.com/goreleaser/goreleaser/internal/pipe/release.doPublish(0xc00035a000, 0x1479ae0, 0xc0000103c8, 0x0, 0x0)
        /root/work/log_rate/goreleaser/internal/pipe/release/release.go:97 +0x29a
github.com/goreleaser/goreleaser/internal/pipe/release.Pipe.Publish(0xc00035a000, 0x0, 0x1de8e30)
        /root/work/log_rate/goreleaser/internal/pipe/release/release.go:83 +0x5c
github.com/goreleaser/goreleaser/internal/middleware.ErrHandler.func1(0xc00035a000, 0xc0003684a0, 0x1e)
        /root/work/log_rate/goreleaser/internal/middleware/error.go:13 +0x33
github.com/goreleaser/goreleaser/internal/middleware.Logging.func1(0xc00035a000, 0x0, 0x0)
        /root/work/log_rate/goreleaser/internal/middleware/logging.go:35 +0x21e
github.com/goreleaser/goreleaser/internal/pipe/publish.Pipe.Run(0xc00035a000, 0x0, 0x1de8e30)
        /root/work/log_rate/goreleaser/internal/pipe/publish/publish.go:62 +0x144
github.com/goreleaser/goreleaser/internal/middleware.ErrHandler.func1(0xc00035a000, 0xc0003681a0, 0x12)
        /root/work/log_rate/goreleaser/internal/middleware/error.go:13 +0x33
github.com/goreleaser/goreleaser/internal/middleware.Logging.func1(0xc00035a000, 0x0, 0x0)
        /root/work/log_rate/goreleaser/internal/middleware/logging.go:35 +0x21e
main.releaseProject.func1(0x0, 0x0)
        /root/work/log_rate/goreleaser/main.go:127 +0x11b
github.com/caarlos0/ctrlc.(*Ctrlc).Run.func1(0xc000078250, 0xc000358020)
        /root/go/pkg/mod/github.com/caarlos0/ctrlc@v1.0.0/ctrlc.go:39 +0x27
created by github.com/caarlos0/ctrlc.(*Ctrlc).Run
        /root/go/pkg/mod/github.com/caarlos0/ctrlc@v1.0.0/ctrlc.go:38 +0x64
```

--- a/internal/client/gitlab.go
+++ b/internal/client/gitlab.go
@@ -68,7 +68,7 @@ func (c *gitlabClient) CreateRelease(ctx *context.Context, body string) (release
 	name := title
 	tagName := ctx.Git.CurrentTag
 	release, resp, err := c.client.Releases.GetRelease(projectID, tagName)
-	if err != nil && resp.StatusCode != 403 {
+	if err != nil && (resp == nil || resp.StatusCode != 403) {
 		return "", err
 	}
 

